builder/ibmcloud/vpc/step_create_vpc_service_instance.go

```diff
@@ -0,0 +1,46 @@
+package vpc
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/IBM/go-sdk-core/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	"github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+type stepCreateVPCServiceInstance struct {
+}
+
+func (step *stepCreateVPCServiceInstance) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	client := state.Get("client").(*IBMCloudClient)
+	ui := state.Get("ui").(packer.Ui)
+	iamurl := os.Getenv("AUTH_URL")
```

Comment:
Avoid input variables from other sources than config.go file. For instance this piece of code should looks like:
@ step_create_vpc_service_instance
```
	config := state.Get("config").(Config)

	ui.Say("Creating VPC service...")

	options := &vpcv1.VpcV1Options{
		Authenticator: &core.IamAuthenticator{
			ApiKey: client.IBMApiKey,
			URL:    config.EndPointIAM,
		},
		URL: config.EndPoint,
	}
```
	
@ config.go
```
type Config struct {
        ....
	EndPointIAM      string `mapstructure-to-hcl2:",skip"`
	EndPoint         string `mapstructure-to-hcl2:",skip"`
	....
	if c.Region == "" {
		errs = packer.MultiErrorAppend(errs, errors.New("a region must be specified"))
	}
       // Configure IBM Cloud EndPoint and EndPointIAM
	c.EndPointIAM = "https://iam.test.cloud.ibm.com/identity/token/"
	// c.EndPoint = "https://" + c.Region + "-stage01.iaasdev.cloud.ibm.com/v1/"
	c.EndPoint = "https://" + c.Region + ".iaas.cloud.ibm.com/v1/"	
```
	
Also note that at this point the EndPoint MUST be production not testing

Reply:
Addressed